commands/command_parser.py

```diff
@@ -20,9 +20,13 @@ class CommandParser:
     
     @classmethod
     def parse_command(cls, bot_query: BotQuery) -> BotCommand:
+        if bot_query.has_callback_query:
+            cmd_name = bot_query.callback_query.callback_data.rsplit('/')[-1].lower()
+            return AVAILABLE_COMMANDS[cmd_name](bot_query) if cmd_name in AVAILABLE_COMMANDS else InvalidCommand(bot_query)
```

Comment:
Nitpick: I'm not personally a fan of this ternary statement because it is way too dense. You can decide if you want to keep this the way it is, but I would have done:
```
if cmd_name:
  return AVAILABLE_COMMANDS[cmd_name](bot_query)
return InvalidCommand(bot_query)
```

Here, control flow is shown through the flow of the statements, instead of being paired with the actual return logic. I generally reserve ternary statements for completely trivial return values (eg. return 1 if is_seen else 0)